### PR TITLE
Use improved TNG farming API

### PR DESCRIPTION
**Pull request by lnjX**

### CODE REVIEW

**Comment by jbruechert** [comment]
I I remember I once planned to port this mod to the TNG API, and now the API got ported to the mod 😀

I'll test & merge it in the evening

File: src/api_crops.lua [hunk 1]
```diff
@@ -0,0 +1,412 @@
...
+			recipe = {
+				{harvestname}
+			}
+		})
+	end
+
+	-- Add items to lists
+	table.insert(agriculture.registered_seeds, seedname)
+end
+
```
Comment (jbruechert) [inline src/api_crops.lua]: .git/rebase-apply/patch:215: trailing whitespace.

.git/rebase-apply/patch:241: trailing whitespace.

.git/rebase-apply/patch:487: trailing whitespace.
 
.git/rebase-apply/patch:487: new blank line at EOF.
+
.git/rebase-apply/patch:864: new blank line at EOF.
+
warning: 5 Zeilen fügen Whitespace-Fehler hinzu.

**Review by jbruechert — CHANGES_REQUESTED** [review]

**Comment by jbruechert** [comment]
```
2019-07-14 17:15:44: ERROR[Main]: ServerError: AsyncErr: ServerThread::run Lua: Runtime error from mod 'agriculture' in callback item_OnPlace(): Runtime error from mod 'agriculture' in callback node_on_construct(): /home/jbb/.minetest/mods/agriculture/src/api_crops.lua:159: attempt to call
2019-07-14 17:15:44: ERROR[Main]:  field 'get_heat' (a nil value)
2019-07-14 17:15:44: ERROR[Main]: stack traceback:
2019-07-14 17:15:44: ERROR[Main]:       /home/jbb/.minetest/mods/agriculture/src/api_crops.lua:159: in function 'start_timer'
2019-07-14 17:15:44: ERROR[Main]:       /home/jbb/.minetest/mods/agriculture/src/api_crops.lua:300: in function </home/jbb/.minetest/mods/agriculture/src/api_crops.lua:299>
2019-07-14 17:15:44: ERROR[Main]:       [C]: in function 'set_node'
2019-07-14 17:15:44: ERROR[Main]:       /home/jbb/.minetest/mods/agriculture/src/api_crops.lua:68: in function </home/jbb/.minetest/mods/agriculture/src/api_crops.lua:21>
2019-07-14 17:15:44: ERROR[Main]: stack traceback:
2019-07-14 17:15:44: ERROR[Main]:       [C]: in function 'set_node'
2019-07-14 17:15:44: ERROR[Main]:       /home/jbb/.minetest/mods/agriculture/src/api_crops.lua:68: in function </home/jbb/.minetest/mods/agriculture/src/api_crops.lua:21>
```

**Comment by lnjX** [comment]
Yes, `core.get_heat()` and `core.get_humidity()` require Mintest 5.0.0+.

**Comment by jbruechert** [comment]
I thought you were still using 0.4.X because that's the version in debian xD

**Comment by lnjX** [comment]
debian experimental ftw!

**Pull request merged**